packages/web3-eth/test/integration/defaults.test.ts

```diff
@@ -0,0 +1,525 @@
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import WebSocketProvider from 'web3-providers-ws';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Contract } from 'web3-eth-contract';
+import { hexToNumber, numberToHex } from 'web3-utils';
+import { TransactionBuilder, TransactionTypeParser, Web3Context } from 'web3-core';
+import { DEFAULT_RETURN_FORMAT } from 'web3-common';
+import { prepareTransactionForSigning, transactionBuilder, Web3Eth } from '../../src';
+
+import { createNewAccount, getSystemTestProvider } from '../fixtures/system_test_utils';
+import {
+	defaultTransactionBuilder,
+	getTransactionFromAttr,
+	getTransactionType,
+} from '../../src/utils';
+import { BasicAbi, BasicBytecode } from '../shared_fixtures/build/Basic';
+import { detectTransactionType } from '../../dist';
+import { getTransactionGasPricing } from '../../src/utils/get_transaction_gas_pricing';
+
+describe('defaults', () => {
+	let web3Eth: Web3Eth;
+	let accounts: string[] = [];
+	let clientUrl: string;
+	let contract: Contract<typeof BasicAbi>;
+	let deployOptions: Record<string, unknown>;
+	let sendOptions: Record<string, unknown>;
+
+	beforeAll(async () => {
+		clientUrl = getSystemTestProvider();
+		const acc1 = await createNewAccount({ unlock: true, refill: true });
+		const acc2 = await createNewAccount({ unlock: true, refill: true });
+		accounts = [acc1.address, acc2.address];
+		web3Eth = new Web3Eth(clientUrl);
+
+		contract = new Contract(BasicAbi, undefined, undefined, web3Eth.getContextObject() as any);
+
+		deployOptions = {
+			data: BasicBytecode,
+			arguments: [10, 'string init value'],
+		};
+
+		sendOptions = { from: accounts[0], gas: '1000000' };
+
+		contract = await contract.deploy(deployOptions).send(sendOptions);
+	});
+	afterAll(() => {
+		if (clientUrl.startsWith('ws')) {
+			(web3Eth.provider as WebSocketProvider).disconnect();
+		}
+	});
+
+	describe('defaults', () => {
+		it('defaultAccount', () => {
+			// default
+			expect(web3Eth.defaultAccount).toBeNull();
+
+			// after set
+			web3Eth.setConfig({
+				defaultAccount: accounts[0],
+			});
+			expect(web3Eth.defaultAccount).toBe(accounts[0]);
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					defaultAccount: accounts[1],
+				},
+			});
+			expect(eth2.defaultAccount).toBe(accounts[1]);
```

Comment:
we should have tests for `defaultAccount` :
A. Set `dafaultAccount` and use none in 'from' property for following functions, and these functions should have 'from' populated with defaultAccount:

- web3.eth.sendTransaction()
- web3.eth.call()
- new web3.eth.Contract() -> myContract.methods.myMethod().call()
- new web3.eth.Contract() -> myContract.methods.myMethod().send()

B. set `dafaultAccount` and also explicitly provide 'from' field for above list of functions, the value provided via 'from' should take priority instead of defaultAccount

Reply:
https://github.com/ChainSafe/web3.js/issues/5117